### ISSUE TITLE
vmstorage -storage.minFreeDiskSpaceBytes option seems does not working

### ISSUE THREAD

**Issue by tr2vil** [VictoriaMetrics/VictoriaMetrics#4023]
### Is your question request related to a specific component?

Victoriametrics cluster version 1.89.1 at vmstorage

### Describe the question in detail

I was using 1.69.0 version of victoriametrics cluster and upgraded to 1.89.1 version today.
I have 6 select servers, 3 insert servers, and 16 storage servers, and I set the -storage.minFreeDiskSpaceBytes option on the storage servers to 130GB. 

In 1.69.0 version, 3 of the storage servers were maintained well with 130GB of free disk space, but after the upgrade, the 130GB free disk space is ignored and data is accumulating continuously. Also 

I have vmstorage turned on with the following options. I was wondering if there is something wrong with my setup.

vmstorage-prod -vminsertAddr=:{insertport_no} -vmselectAddr=:{selectport_no} \.
-retentionPeriod=1095d \
-storageDataPath=/data/victoria_metrics \
-httpListenAddr=:{httpport_no} \
-loggerFormat=json \
-loggerLevel=INFO \
-dedup.minScrapeInterval=60s \
-http.connTimeout=1m \
-search.maxUniqueTimeseries=900000 \
-finalMergeDelay=0 \
-storage.minFreeDiskSpaceBytes=130GB \
-memory.allowedPercent=60 \
-search.maxConcurrentRequests=7 \
-search.maxQueueDuration=8s \
-search.maxTagKeys=90000 \
-search.maxTagValues=90000

Additionally, is the amount of disk occupied by indexdb/file not governed by the -storage.minFreeDiskSpaceBytes setting? I see now that the data size of the indexdb path is as follows.

```
# du -hs *.
173G indexdb
```

Finally, I can see in the log that vmstorage fell into read-only mode but my /data disk is fulled.
```
# df -h                                                                                                                                                                         
Filesystem      Size  Used Avail Use% Mounted on

/dev/sda7        20G  1.6G   17G   9% /home
/dev/sda8       1.7T  1.6T   11G 100% /data
```

### Troubleshooting docs

- [X] General - https://docs.victoriametrics.com/Troubleshooting.html
- [ ] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by hagen1778** [comment]
Hi @tr2vil!
Do you have metric `vm_free_disk_space_bytes` scraped? Can you show how it changed after the update and until the storage node became read-only?

**Comment by tr2vil** [comment]
Sir @hagen1778. 

Out of a total of 16 storage servers, 
the 3 servers in question today were servers that had previously accumulated data up to minFreeDiskSpaceBytes. 
The data was not increasing according to the minFreeDiskSpaceBytes setting. 
I performed the upgrade at 7am today from older version(1.66.2) to newer version(1.89.1) and have been restarting the STORAGE nodes from time to time as the disks fill up, so the graph is a bit choppy, but it looks like this

![image](https://user-images.githubusercontent.com/5829306/228193916-0924681e-e34b-4518-b6fe-2ab5ce19aa50.png)

First of all, I rolled back to the original version at 6pm and the data is still not increasing.

**Comment by hagen1778** [comment]
Thanks for the info!
Do I understand correct the unit on Y axis is billions and it says before the update it was 300GB of free disk space? If yes, how that could be if `-storage.minFreeDiskSpaceBytes=130GB` was set?

May I also ask you to plot the following graphs for those 3 storage nodes:
* vm_storage_is_read_only
* vm_rpc_vmstorage_is_read_only
Thanks!

**Comment by tr2vil** [comment]
각각의 storage 서버는 2TB 디스크를 가지고 있었으며, -storage.minFreeDiskSpaceBytes=300GB 으로 운영을 했었습니다. 금일 업데이트 이후 가용 디스크 공간이 급격히 줄어들어 그 양을 300에서 130까지 낮춰보았던 것 입니다.

말씀하신 그래프는 다음과 같습니다.
![image](https://user-images.githubusercontent.com/5829306/228200595-60c533aa-b2f3-49c0-b65e-e70c8f04837e.png)

vm_rpc_vmstorage_is_read_only is not queried.

**Comment by hagen1778** [comment]
Weird...
Can you show `sum(vm_data_size_bytes) by (instance, type)` for those 3 storage nodes?
And if you have Grafana dashboard for cluster version - screenshot of Troubleshooting section?

**Comment by tr2vil** [comment]
Hi @hagen1778 , First of all, I'm in Korea and apologize for the slow response due to the time difference.

I checked what you said and it's strange.
The result of sum(vm_data_size_bytes) by (instance, type) looks horizontal.

![image](https://user-images.githubusercontent.com/5829306/228393970-ec6dad59-ef1f-4030-a03c-2ff6cc3c9dd4.png)


The screenshot of Troubleshooting section is listed below.
![image](https://user-images.githubusercontent.com/5829306/228394161-6f4a6efb-b4f0-4840-80b4-64bd851773ad.png)

**Comment by tr2vil** [comment]
Thanks for you help. @hagen1778 

Based on your mention(#4035), does this mean that the feature could be merged into the next version?

**Comment by valyala** [comment]
The issue should be fixed in [v1.87.5 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.5).

@tr2vil , please re-open the issue if it isn't fixed yet for your workload in `v1.87.5` release.

**Comment by valyala** [comment]
FYI, the bugfix has been also included in [v1.79.12 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.12).

**Comment by valyala** [comment]
FYI, the bugfix is included also in all the VictoriaMetrics releases starting from [v1.90.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
